### PR TITLE
Plugin Management: Refactor actions and statuses

### PR DESCRIPTION
**Pull request by mattgawarecki**
This pull request contains only the refactoring changes that were added in #80281. See that PR for further info.

Is depended on by #80281.
Related to `1202619025189113-as-1202979359232070`.

## Proposed Changes

* Additions:
  * Create a new hook `useStatusMessageText`, to render action status messages.
  * Create a new hook `useEmptyMessage`, to render messages when no plugins are found.
* Changes:
  * Rename the `RenderStatusMessage` component to `StatusMessage`.
  * Replace usages of `getPluginActionStatusMessages` with `useStatusMessageText`.
  * Replace usages of `pluginsEmptyMessage` with `useEmptyMessage`.
  * Replace usages of `renderBulkActionsHeader` with `BulkActionsHeader`.
  * Extract all action status messages into their own files (in `plugin-action-status/messages`), grouped by the action type.
* Removals:
  * the `renderBulkActionsHeader` function from `PluginManagementV2` (in favor of the `BulkActionsHeader` component)
  * `pluginsEmptyMessage` from `utils.ts` (in favor of `useEmptyMessage`)
  * `getPluginActionStatusMessages` from `utils.ts` (in favor of `useStatusMessageText`)

## Testing Instructions

* In your test environment, with this PR applied, visit `/plugins/manage` in either Calypso Blue or Calypso Green.
* Take various actions to manage plugins on one or more of your sites. Verify that all messages appear correctly, match the action you selected, and that the status of the selected plugins matches what you expect while the actions are performed (e.g., "Activating," then "Activated"; or, "Deactivating on 4 sites"; "Deactivated on 4 sites")
* Inspect the PR's code changes, asking plenty of questions to ensure the changes all make sense.

### CODE REVIEW

File: client/my-sites/plugins/plugin-management-v2/index.tsx [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+		if ( searchTerm ) {
+			return translate( 'No results found. Please try refining your search.' );
+		}
+
+		const getMessage = (
+			{
+				all: ( translate ) => translate( 'No plugins found.' ),
+				active: ( translate ) => translate( 'No plugins are active.' ),
+				inactive: ( translate ) => translate( 'No plugins are inactive.' ),
+				updates: ( translate ) => translate( 'All plugins are up to date.' ),
```
Comment (yashwin) [inline client/my-sites/plugins/plugin-management-v2/index.tsx:30]: Why are we passing `translate` as an argument here?

File: client/my-sites/plugins/plugin-management-v2/plugin-action-status/messages/install.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import {
+	PLUGIN_INSTALLATION_COMPLETED as COMPLETED,
+	PLUGIN_INSTALLATION_ERROR as ERROR,
+	PLUGIN_INSTALLATION_IN_PROGRESS as IN_PROGRESS,
+} from 'calypso/state/plugins/installed/status/constants';
+import type { PluginActionMessagesByStatus } from './types';
+
+const InstallMessageGetters: PluginActionMessagesByStatus = {
```
Comment (yashwin) [inline client/my-sites/plugins/plugin-management-v2/plugin-action-status/messages/install.ts]: Can we name this as `InstallMessages` to keep the names consistent?

File: client/my-sites/plugins/plugin-management-v2/plugin-action-status/messages/update.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import {
+	PLUGIN_INSTALLATION_COMPLETED as COMPLETED,
+	PLUGIN_INSTALLATION_ERROR as ERROR,
+	PLUGIN_INSTALLATION_IN_PROGRESS as IN_PROGRESS,
+	PLUGIN_INSTALLATION_UP_TO_DATE as UP_TO_DATE,
+} from 'calypso/state/plugins/installed/status/constants';
+import type { PluginActionMessagesByStatus } from './types';
+
+const ActivateMessages: PluginActionMessagesByStatus = {
```
Comment (yashwin) [inline client/my-sites/plugins/plugin-management-v2/plugin-action-status/messages/update.ts]: Can we name this `UpdateMessages`?

File: client/my-sites/plugins/plugin-management-v2/plugin-action-status/messages/activate.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import {
+	PLUGIN_INSTALLATION_COMPLETED as COMPLETED,
+	PLUGIN_INSTALLATION_ERROR as ERROR,
+	PLUGIN_INSTALLATION_IN_PROGRESS as IN_PROGRESS,
+} from 'calypso/state/plugins/installed/status/constants';
+import type { PluginActionMessagesByStatus } from './types';
+
+const ActivateMessageGetters: PluginActionMessagesByStatus = {
```
Comment (yashwin) [inline client/my-sites/plugins/plugin-management-v2/plugin-action-status/messages/activate.ts]: Can we name this `ActivateMessages` to keep it consistent with other files?

**Review by yashwin — CHANGES_REQUESTED** [review]
Everything works as expected.

However, I have left a few comments.

File: client/my-sites/plugins/plugin-management-v2/index.tsx (hunk 1 above)
Comment (mattgawarecki) [reply client/my-sites/plugins/plugin-management-v2/index.tsx:30]: It's not strictly necessary, but it was my attempt at a performance optimization. By deferring the translation until we know for sure which message we need, we're saving ourselves 4 calls to `translate` each time this hook gets called.

More concretely:
- If we directly translate `all`, `active`, `inactive`, and `updates`, we have an object with translations for all four messages, regardless of the value of `filter`.
- If we write each message as a function that takes `translate` as an argument, nothing actually gets translated until the call to `getMessage?.( translate )` on line 36. And at that point, only the one message we need to display is translated.

**Review by yashwin — APPROVED** [review]
LGTM 🚀

**Pull request merged**